olp-cpp-sdk-core/src/cache/DefaultCacheImpl.cpp

```diff
@@ -832,9 +833,17 @@ DefaultCache::StorageOpenResult DefaultCacheImpl::SetupProtectedCache() {
   StorageSettings protected_storage_settings;
   protected_storage_settings.max_file_size = 32 * 1024 * 1024;
 
-  auto status = protected_cache_->Open(
-      settings_.disk_path_protected.get(), settings_.disk_path_protected.get(),
-      protected_storage_settings, OpenOptions::ReadOnly);
+  OpenOptions open_mode = settings_.openOptions;
+  if (!(open_mode & OpenOptions::ReadOnly)) {
+    if (utils::Dir::IsReadOnly(settings_.disk_path_protected.get() +
+                               "\\LOCK")) {
```

Comment:
Magic strings are never good. Can we not check on the directory instead and assume it is the same for all included files?

Reply:
This is a test code that sneaked in. Removed that and just checking directory.